acsylla/_cython/cluster/cluster.pyx

```diff
@@ -1,22 +1,80 @@
+import socket
+
+from posix cimport unistd
+
+
 cdef class Cluster:
+    def destroy(self):
+        if self.host_listener:
+            self.host_listener.destroy()
+            self.host_listener = None
+        if self.logger:
+            self.logger.destroy()
+            self.logger = None
+        if self.loop and self.loop.is_running() and self._read_socket:
+            self.loop.remove_reader(self._read_socket)
+            self.loop = None
+            self._read_socket.close()
+            self._read_socket = None
+        if self._write_socket:
+            self._write_socket.close()
+            self._write_socket = None
+
+    def __dealloc__(self):
+        cass_cluster_free(self.cass_cluster)
+        if self.ssl != NULL:
+            cass_ssl_free(self.ssl)
+        self.destroy()
 
     def __cinit__(self):
-        # Starts the necessary machinary for bringing events
-        # from the CPP driver to a Python Thread.
-        # Is idempotent, can be called as many times as we want
-        # but would be initalize only once.
-        _initialize_posix_to_python_thread()
```

Comment:
Seems that we are moving some if not all of the logic behind https://github.com/acsylla/acsylla/blob/43d228de3f1cef2a55f98bfffa14e10c6e195c7c/acsylla/_cython/posix_to_python_thread.pyx under the `Cluster` class, is that correct?

If so, wondering if after this PR will be some dead code in the `_cython/posix_to_python_thread.pyx` that could be remove.

Reply:
> @pfreixes this is an experiment to be able to run a cluster in different threads with different event loops

Can we isolate changes, so we start with a PR for making it more pythonic and we iterate later with this experiment? we can also decouple this experiment form having the version 1.x published

Reply:
This is no longer an experiment but a stable working code: I rewrote see https://github.com/acsylla/acsylla/pull/95/commits/61e4b6ca14aea64cf1273094c3aaa2b13e993fef ;)